pkg/gadgets/internal/socketenricher/bpf/sockets-map.h

```diff
@@ -81,26 +81,38 @@ struct {
 static __always_inline struct sockets_value *
 gadget_socket_lookup(const struct __sk_buff *skb)
 {
-	// Only IPv4 is supported for now
-	if (load_half(skb, offsetof(struct ethhdr, h_proto)) != ETH_P_IP)
-		return 0;
-
 	struct sockets_key key = {0,};
+	int l4_off;
+
 	key.netns = current_netns;
-	key.family = AF_INET;
-	key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
+	key.family = load_half(skb, offsetof(struct ethhdr, h_proto));
```

Comment:
nit: use another variable? It's a bit confusing to have the same variable to store the ether proto and the family.

Reply:
It is still 2 variables for 2 different things:
1. key.family: layer 3 protocol (aka family): IPv4 or IPv6
2. key.proto: layer 4 protocol: tcp or udp

My PR didn't change that. The patch might be confusing because I moved the assignment of `key.proto` below.

I could rename the fields to `l3_proto` and `l4_proto` to avoid confusion? (in another PR)

Reply:
My point is, in L88 it's reading **`h_proto`** (i.e. [EtherType](https://en.wikipedia.org/wiki/EtherType)), but then it's being converted to family, AF_INET in L91 and AF_INET6 in L96. I find it a bit confusing that h_proto is saved in a variable named "family", when those are two different things. 

That change I propose is:

```diff
$ git diff
diff --git pkg/gadgets/internal/socketenricher/bpf/sockets-map.h pkg/gadgets/internal/socketenricher/bpf/sockets-map.h
index 935d3502..f68ef38a 100644
--- pkg/gadgets/internal/socketenricher/bpf/sockets-map.h
+++ pkg/gadgets/internal/socketenricher/bpf/sockets-map.h
@@ -83,10 +83,11 @@ gadget_socket_lookup(const struct __sk_buff *skb)
 {
        struct sockets_key key = {0,};
        int l4_off;
+       __u16 h_proto
 
        key.netns = current_netns;
-       key.family = load_half(skb, offsetof(struct ethhdr, h_proto));
-       switch (key.family) {
+       h_proto = load_half(skb, offsetof(struct ethhdr, h_proto));
+       switch (h_proto) {
        case ETH_P_IP:
                key.family = AF_INET;
                key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
```